subiquity/server/controllers/identity.py

```diff
@@ -16,15 +16,68 @@
 import logging
 
 import attr
+import os
+import re
+from typing import Set
 
 from subiquitycore.context import with_context
 
 from subiquity.common.apidef import API
-from subiquity.common.types import IdentityData
+from subiquity.common.resources import resource_path
+from subiquity.common.types import IdentityData, UsernameValidation
 from subiquity.server.controller import SubiquityController
 
 log = logging.getLogger('subiquity.server.controllers.identity')
 
+USERNAME_MAXLEN = 32
+USERNAME_REGEX = r'[a-z_][a-z0-9_-]*'
+
+
+def _reserved_names_from_file(path: str) -> Set[str]:
+    names = set()
+    try:
+        with open(path, "r") as f:
+            for line in f:
+                line = line.strip()
+                if not line or line.startswith('#'):
+                    continue
+                names.add(line.split()[0])
+
+    except FileNotFoundError:
```

Comment:
I'm concerned about only logging these failures.  Failure to process the reserved-names file to me represents an installer bug and I think we should allow those Exceptions to bubble up to the surface.